### DISCUSSION TITLE
Is there a way to make qutebrowser behave like firefox with the 'Query Parameter Stripping'  function enabled?

### DISCUSSION THREAD

**Discussion by amalgame21** [qutebrowser/qutebrowser#7684] [General] [answered]
I would like to find a way to make qutebrowser bahaves like firefox with the 'Query Parameter Stripping' `privacy.query_stripping.enabled.pbmode` function enabled in `about:config`, where we can strip the query strings like `fbclid` and `utm_campaign` after pressing enter in the url bar or click on a link. 

I found a script in https://github.com/qutebrowser/qutebrowser/issues/5737#issue-705754077 which is basically what I want, but it can only strip the `fbclid` out, not others. 
I have no python knowledge, so I failed to modified the script to make it work with other parameters. 

Any existing method can achieve this? thanks!
By the way, I just know that we can yank the url to ignore custom paramter with `url.yank_ignored_parameters`, which is very nice!

**Comment by toofar** [comment, marked as answer]
> Any existing method can achieve this? 

I don't think so. Just the interceptor API you've already seen. Here's my attempt at adapting that one you posted. Note that I haven't actually ran this, but hopefully the logic is clear
```python

from qutebrowser.api import interceptor
from qutebrowser.qt.core import QUrl, QUrlQuery

config.load_autoconfig()

naughty_params = ["fbclid", "utm_campaign"]
def intercept(info: interceptor.Request) -> None:
    if info.request_url.hasQuery():
        has_changes = False
        query = QUrlQuery(info.request_url.query())

        for param in naughty_params:
            if query.hasQueryItem(param):
                has_changes = True
                query.removeQueryItem(param)

        if has_changes:
            new_url = QUrl(info.request_url)
            new_url.setQuery(query)
            print(new_url)
            try:
                info.redirect(new_url, ignore_unsupported=True)
            except interceptor.interceptors.RedirectException as err:
                print(err)

interceptor.register(intercept)
```
Edit: added exception handling :sunglasses:

**Reply by amalgame21** [reply]
Thanks! Just replaced `qutebrowser.qt.core` with `PyQt5.QtCore` (maybe `PyQt6.QtCore` later?), and it ran just perfectly fine!! Thank you for your help!!!

**Reply by amalgame21** [reply]
I foung that when the  `__s` naughty_params is included (it is a "Drip.com email address tracking parameter" according to privacytests.org), it 100% crash qutebrowser with all the facebook links. Do you have any clue?

**Reply by The-Compiler** [reply]
Most likely an upstream issue - please open an issue with detailed version information and reproduction steps.
